File: frontend/lib/types/project.ts
export type VideoType =
  | "product_explainer"
  | "educational"
  | "agent_training"
  | "social_ad"
  | "testimonial"
  | "corporate_update"
  | "compliance";

export type ProjectStatus = "draft" | "in_review" | "exported";

export const VIDEO_TYPE_LABELS: Record<VideoType, string> = {
  product_explainer: "Product Explainer",
  educational: "Educational / Awareness",
  agent_training: "Agent Training",
  social_ad: "Social Media Ad",
  testimonial: "Customer Testimonial",
  corporate_update: "Corporate Update",
  compliance: "Compliance",
};

export const BRAND_KITS = [
  "SecureLife Standard (Warm, Professional, Blue)",
  "SecureLife Social (Friendly, Casual, Teal)",
  "No brand constraints",
];

export const TARGET_DURATIONS = [
  { value: 15, label: "15 seconds (2–3 scenes)" },
  { value: 30, label: "30 seconds (3–4 scenes)" },
  { value: 60, label: "60 seconds (5 scenes)" },
  { value: 90, label: "90 seconds (6–8 scenes)" },
];

export interface Project {
  id: string;
  user_id: string;
  title: string;
  video_type: VideoType | null;
  target_audience: string | null;
  target_duration: number | null;
  key_message: string | null;
  brand_kit: string | null;
  call_to_action: string | null;
  script: string | null;
  tone: string | null;
  word_count: number | null;
  status: ProjectStatus;
  presenter_id: string | null;
  presenter: import("./presenter").Presenter | null;
  scenes: import("./scene").Scene[];
  videos: import("./video").Video[];
  created_at: string;
  updated_at: string;
}

export interface ProjectStats {
  total: number;
  drafts: number;
  exported: number;
  in_review: number;
  videos_generated: number;
}

File: backend/app/schemas/project.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional, Literal
from app.models.project import VideoType, ProjectStatus
from app.schemas.presenter import PresenterResponse
from app.schemas.scene import SceneResponse
from app.schemas.video import VideoResponse


class ProjectCreate(BaseModel):
    title: str
    video_type: Optional[VideoType] = None
    target_audience: Optional[str] = None
    target_duration: Optional[int] = None
    key_message: Optional[str] = None
    brand_kit: Optional[str] = None
    call_to_action: Optional[str] = None
    presenter_id: Optional[UUID] = None


class ProjectUpdate(BaseModel):
    title: Optional[str] = None
    video_type: Optional[VideoType] = None
    target_audience: Optional[str] = None
    target_duration: Optional[int] = None
    key_message: Optional[str] = None
    brand_kit: Optional[str] = None
    call_to_action: Optional[str] = None
    script: Optional[str] = None
    tone: Optional[str] = None
    word_count: Optional[int] = None
    status: Optional[ProjectStatus] = None
    presenter_id: Optional[UUID] = None


class ProjectResponse(BaseModel):
    id: UUID
    user_id: UUID
    title: str
    video_type: Optional[VideoType]
    target_audience: Optional[str]
    target_duration: Optional[int]
    key_message: Optional[str]
    brand_kit: Optional[str]
    call_to_action: Optional[str]
    script: Optional[str]
    tone: Optional[str]
    word_count: Optional[int]
    status: ProjectStatus
    presenter_id: Optional[UUID]
    presenter: Optional[PresenterResponse]
    scenes: list[SceneResponse] = []
    videos: list[VideoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    id: UUID
    user_id: UUID
    title: str
    video_type: Optional[VideoType]
    target_duration: Optional[int]
    status: ProjectStatus
    presenter_id: Optional[UUID]
    presenter: Optional[PresenterResponse]
    scene_count: int = 0
    video_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectStatsResponse(BaseModel):
    total: int
    drafts: int
    exported: int
    in_review: int
    videos_generated: int


class ExportRequest(BaseModel):
    format: Literal["full_package", "script_only", "json"]
